build-system/tasks/performance/measure-documents.js

```diff
@@ -118,6 +196,124 @@ const readMetrics = (page) =>
     };
   });
 
+/**
+ * Set up defaults handlers for docs that will be requested
+ * that are not explicted stated as script tags and not
+ * handled by the 'additionalHandlers'.
+ *
+ * @param {!Array<function>} handlersList
+ * @param {string} version
+ */
+function setupDefaultHandlers(handlersList, version) {
+  // Handle requests made by runtime for default extensions
+  handlersList.push((interceptedRequest) =>
+    defaultExtensionsHandler(interceptedRequest, version)
+  );
+}
+
+/**
+ * Set up appropriate handlers based upon handlerOptions
+ *
+ * @param {!Array<function>} handlersList
+ * @param {?Object} handlerOptions
+ * @param {!Puppeteer.page} page
+ * @param {!function} resolve
+ */
+async function setupAdditionalHandlers(
+  handlersList,
+  handlerOptions,
+  page,
+  resolve
+) {
+  switch (handlerOptions.handlerName) {
+    case 'analyticsHandler':
+      // Set up timing
+      Object.assign(handlerOptions, {'startTime': Date.now()});
+      handlersList.push((interceptedRequest) =>
```

Comment:
are we returning a promise here with analyticsHandler?

Reply:
Adding the `analyticsHandler` to the `handlerList`. The `analyticsHandler` returns a promise to signal if it actually did anything with the request.